plugins/inputs/system/ps.go

```diff
@@ -73,35 +73,27 @@ func (s *systemPS) DiskUsage(
 	var usage []*disk.UsageStat
 	var partitions []*disk.PartitionStat
 
-	for i := range parts {
-
-		p := parts[i]
+	for _, p := range parts {
+		// If the mount point is not a member of the filter set,
+		// don't gather info on it.
+		if _, ok := mountPointFilterSet[p.Mountpoint]; !ok {
+			continue
+		}
 
-		if len(mountPointFilter) > 0 {
-			// If the mount point is not a member of the filter set,
-			// don't gather info on it.
-			_, ok := mountPointFilterSet[p.Mountpoint]
-			if !ok {
-				continue
-			}
+		// If the mount point is a member of the exclude set,
+		// don't gather info on it.
+		if _, ok := fstypeExcludeSet[p.Fstype]; ok {
+			continue
 		}
+
 		mountpoint := os.Getenv("HOST_MOUNT_PREFIX") + p.Mountpoint
 		if _, err := os.Stat(mountpoint); err == nil {
-			du, err := disk.Usage(mountpoint)
-			if err != nil {
-				return nil, nil, err
-			}
-			du.Path = p.Mountpoint
-
-			// If the mount point is a member of the exclude set,
-			// don't gather info on it.
-			_, ok := fstypeExcludeSet[p.Fstype]
-			if ok {
-				continue
+			if du, err := disk.Usage(mountpoint); err == nil {
+				du.Path = p.Mountpoint
+				du.Fstype = p.Fstype
```

Comment:
It's because of things like this:

```
rootfs / rootfs rw 0 0
/dev/mapper/centos-root / xfs rw,relatime,attr2,inode64,noquota 0 0
```

The `for ... range parts {` is going to iterate over both these. However when the `disk.Usage()` call runs for the first one, it's going to get the second one because it's mounted on top. So `p.Fstype == "rootfs"` but `du.Fstype == "xfs"`.